src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -689,7 +673,75 @@ protected bool AccessorsHaveImplementation
         /// a property with an accessor using the 'field' keyword, or
         /// a property with an initializer.
         /// </summary>
-        internal SynthesizedBackingFieldSymbol BackingField { get; }
+        internal SynthesizedBackingFieldSymbol BackingField
+        {
+            get
+            {
+                if (_lazyMergedBackingField is null)
+                {
+                    var backingField = DeclaredBackingField;
+                    if (!_containingType.AreMembersComplete)
+                    {
+                        // The property should only be used after members in the containing
+                        // type are complete, and partial members have been merged.
+                        return backingField;
```

Comment:
I think we should also assert that we are in one of the scenarios which is known to have this problem (containing type has primary constructor).

Reply:
I don't think we should assert here since this is a property that might be invoked in the debugger, before containing members are complete.

Reply:
Could we assert `!Debugger.IsAttached`?